architecture/spmc.rst

```diff
@@ -0,0 +1,267 @@
+====
+SPMC
+====
+
+This document describes the SPMC (S-EL1) implementation for OP-TEE.
+More information on the SPMC can be found in the FF-A specification can be
+found in the
+`FF-A spec <https://developer.arm.com/documentation/den0077/latest>`_.
+
+.. toctree::
+ :numbered:
+
+SPMC Responsibilities
+=====================
+
+The SPMC is a critical component in the FF-A flow. Some of it mayor
+responsibilities are:
+
+- Initialisation and run-time management of the SPs:
+	The SPMC component is responsible for initialisation of the
+	Secure Partitions (Loading the Image, Setting up the stack, HEAP, ...).
+- Routing messages between endpoints:
+	The SPMC is responsible for passing FF-A messages from The Normal World
+	to SPs and back. It also needs to to pass FF-A messages between SPs.
+- Memory management:
+	The SPMC is responsible for the memory management of the SPs. Memory
+	can be shared between SPs and from a SP to the Normal World.
+
+This document describes OP-TEE as a S-EL1 SPMC.
+
+Secure Partitions
+=================
+Secure Partitions (SPs) are the endpoints used in the FF-A protocol. When
+OP-TEE is used as a SPMC SPs run primarily inside S-EL0.
+
+The SPMC also exposes the OP-TEE core, privileged mode, as an secure endpoint
+itself. This is used to handle all GlobalPlaform programming mode operations
```

Comment:
add a dot `.` and end of the above line.

Is the sentence below true? I understand it is a goal but I think this is not yet implemented. If so, maybe this doc should not be stated that as something that is available.

Reply:
This was implemented by @jenswi-linaro

Reply:
ok